### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**

### CODE REVIEW

File: app/router/student.py [hunk 1]
```diff
@@ -125,3 +130,79 @@ async def verify_student(request: Request, student_id: str):
...
         return True
     return HTTPException(status_code=response.status_code, detail=response.errors)
+
+
+def get_jnv_code(data):
+    response = requests.get(
+        school_db_url, params={"region": data.region, "name": data.school_name}
+    )
+    if response.status_code == 200:
+        if len(response.json()) == 0:
```
Comment (suryabulusu) [inline app/router/student.py:140]: in what case will response status_code be 200, but `len(response.json() == 0` be true?

File: app/router/student.py [hunk 2]
```diff
@@ -125,3 +130,79 @@ async def verify_student(request: Request, student_id: str):
...
+            return HTTPException(
+                status_code=404, detail="JNV or region does not exist!"
+            )
+        return response.json()
+    return HTTPException(status_code=response.status_code, detail=response.errors)
+
+
+def generate_three_digit_code(code):
+    for _ in range(3):
+        code += random.randint(0, 9)
```
Comment (suryabulusu) [inline app/router/student.py:150]: `code += str(random.randint(0, 9))`

Also write an example in function comments maybe? 
`"""if input code = '234', output = '234910'"""`

File: app/router/student.py [hunk 3]
```diff
@@ -125,3 +130,79 @@ async def verify_student(request: Request, student_id: str):
...
+                    return student_id
+
+                is_student_added, number_of_loops = False, 0
+                if not is_student_added or number_of_loops < 1000:
+                    if not does_generated_id_exist:
+                        is_student_added = True
+                        # add ID to db
+                        return student_id
+                    else:
+                        student_id[:-3] += generate_three_digit_code("")
```
Comment (suryabulusu) [inline app/router/student.py:200]: we could just have the function as `generate_three_digit_code()` without any input na

File: app/router/student.py [hunk 4]
```diff
@@ -125,3 +130,79 @@ async def verify_student(request: Request, student_id: str):
...
+            if data.group == "EnableStudents":
+                student_id += form_details.grade_code[data.grade]
+                student_id += get_jnv_code(data)  # test
+                student_id += generate_three_digit_code("")
+                does_generated_id_exist = verify_student(student_id=student_id)
+                if does_generated_id_exist:
+                    return student_id
+
+                is_student_added, number_of_loops = False, 0
+                if not is_student_added or number_of_loops < 1000:
```
Comment (suryabulusu) [inline app/router/student.py:194]: where are you using `number_of_loops`?

Comment (suryabulusu) [reply app/router/student.py:194]: I tihnk some for loop should run -- and in each loop we generate a new 3-digit code, and see if ID exists in DB, right? 
Can you also link JS code in PR description? `studentSignup.js` i think

File: app/router/student.py (hunk 1 above)
Comment (sudheshna-donthineni) [reply app/router/student.py:140]: When there is no matching entry found in DB.

**Pull request closed without merging**